src/System.Collections.NonGeneric/src/System/Collections/KeyValuePairs.cs

```diff
@@ -14,14 +14,14 @@
 
 namespace System.Collections
 {
-    [DebuggerDisplay("{value}", Name = "[{key}]", Type = "")]
+    [DebuggerDisplay("{Value}", Name = "[{Key}]", Type = "")]
     internal class KeyValuePairs
```

Comment:
Thanks for fixing this as it was truly a bug. I would however change this to be the field names _value and _key as that will be cheaper to execute in the debugger. While this is a simple property it still requires a function call in the Expression Evaluator which is more expensive than a field look-up.